src/Classes/ExtBuildListControl.lua

```diff
@@ -0,0 +1,551 @@
+			(cursorX < importButton.x1) and
+			(cursorY > importButton.y0) and
+			(cursorY < importButton.y1) then
+			self:HandleButtonClick(importButton, 'import')
+			break
+		end
+	end
+
+	for i, previewButton in ipairs(self.previewButtons) do
+		if (cursorX > previewButton.x0) and
+			(cursorX < previewButton.x1) and
+			(cursorY > previewButton.y0) and
+			(cursorY < previewButton.y1) then
+			self:HandleButtonClick(previewButton, 'preview')
+			break
+		end
+	end
+end
+
+-- splits strings by word and maxWidth
+function ExtBuildListControlClass:splitStringByWidth(str, maxWidth, font)
+	local words = {}
+	for word in str:gmatch("%S+") do
+		t_insert(words, word)
+	end
+
+	local lines = {}
+	local currentLine = ""
+	for _, word in ipairs(words) do
+		local wordWidth = DrawStringWidth(16, font, currentLine .. " " .. word)
+		if wordWidth <= maxWidth then
+			currentLine = currentLine .. (currentLine == "" and "" or " ") .. word
+		else
+			t_insert(lines, currentLine)
+			currentLine = word
+		end
+	end
+	t_insert(lines, currentLine)
+
+	return lines
+end
+
+-- wrappers for Drawing tools to apply scrolling
+function ExtBuildListControlClass:DrawImage(imgHandle, left, top, width, height)
+	local _, y = self:GetPos()
+	if top - self.controls.scrollBarV.offset >= y and top + height - self.controls.scrollBarV.offset < self.height() + y then
+		DrawImage(imgHandle, left, top - self.controls.scrollBarV.offset, width, height)
+	end
+end
+
+function ExtBuildListControlClass:DrawString(left, top, align, height, font, text)
+	local _, y = self:GetPos()
+	if top - self.controls.scrollBarV.offset >= y and top + height - self.controls.scrollBarV.offset < self.height() + y then
+		DrawString(left, top - self.controls.scrollBarV.offset, align, height, font, text)
+	end
+end
+
+function ExtBuildListControlClass:Draw(viewPort, noTooltip)
+	if self.activeListProvider == nil then
+		return
+	end
+
+	-- clear button states
+	wipeTable(self.previewButtons)
+	wipeTable(self.importButtons)
+
+	-- get variables
+	local x, y = self:GetPos()
+	local width, height = self:GetSize()
+
+	-- drawing area
+	-- SetDrawColor(0.5, 0.5, 0.5)
+	SetDrawColor(1, 1, 1)
+	DrawImage(nil, x, y, width, height)
+	-- borders
+	SetDrawColor(0, 0, 0)
+	DrawImage(nil, x + 1, y + 1, width - 2, height - 2)
+	SetDrawColor(1, 1, 1)
+
+	-- apply padding
+	x = x + 10;
+	y = y + 10;
+
+	local currentHeight = y
+
+	-- write status message
+	if self.activeListProvider.statusMsg then
+		self:DrawString(x, currentHeight, "LEFT", 16, self.font, self.activeListProvider.statusMsg)
+	end
+
+	local scrollBarV = self.controls.scrollBarV
+	self.scrollOffsetV = scrollBarV.offset
+
+	-- loop through builds
+	for _, build in pairs(self.activeListProvider.buildList) do
+		if build.buildName then
+			local image = nil
+			if build.ascendancy then
+				image = self:GetAscendancyImageHandle(build.ascendancy)
+			else if build.class then
+				image = self:GetClassImageHandle(build.class)
+				end
+			end
```

Comment:
Simplified:

```suggestion
			if build.ascendancy or build.class then
				image = self:GetImageHandle(build.ascendancy or build.class)
			end
```
Also, just for future reference:
```suggestion
                        if build.ascendancy then
				image = self:GetAscendancyImageHandle(build.ascendancy)
			elseif build.class then
				image = self:GetClassImageHandle(build.class)
			end
```

Reply:
Fixed with the previous commit.